### DISCUSSION TITLE
Dynamic ResourceLock values

### DISCUSSION THREAD

**Discussion by roded** [junit-team/junit-framework#3740] [Q&A] [answered]
Hi,
Is it somehow currently possible to use a calculated value as a resource lock name for a test.
Being a Java annotation, the existing `@ResourceLock` will only accept a static value.

Thanks

**Comment by marcphilipp** [comment]
No that's currently not possible. What would be your use case?

**Reply by roded** [reply]
We run our CI suite on different environments.
Each environment has its own and separate set of resource name.
These resources are loaded during in the static context so addressing them in the ResourceLock annotation isn't possible.
We might be able to solve this with indirection. I'm not currently sure.

**Reply by roded** [reply]
Related: https://github.com/junit-team/junit5/issues/3442#issuecomment-1822711647

**Comment by marcphilipp** [comment, marked as answer]
There's an issue for this: #2677
